### PR TITLE
[functions] Add `ipAddress` & `geolocation`

### PR DESCRIPTION
**Pull request by Kikobeats**
Ported from https://github.com/vercel/vercel/blob/main/packages/edge/src/edge-headers.ts

These methods are not exclusively from edge anymore, they are isomorphic!

Also, added docs generation.

### CODE REVIEW

**Review by ecklf — APPROVED** [review]

File: packages/functions/test/integration/docs.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { expect, test } from 'vitest';
+import $ from 'tinyspawn';
+import path from 'path';
+
+(process.platform === 'win32' ? test.skip : test)(
+  'docs are up to date',
+  async () => {
+    const cwd = path.resolve(__dirname, '../');
+    await $('pnpm build:docs', { cwd });
+    const { stdout } = await $(`git status --short docs`);
```
Comment (ecklf) [inline packages/functions/test/integration/docs.test.ts:10]: This only shows the current changes / working tree. Wouldn't this be always be empty when running in CI?

Comment (Kikobeats) [reply packages/functions/test/integration/docs.test.ts:10]: I tested manually and if you forgot to commit docs it will throw an error!

**Comment by maxkostow** [comment]
Is this intended to work with node `IncomingMessage` based next functions? Those do not have `get` defined on the `headers` object

**Comment by wyattjoh** [comment]
This will work with Next.js's App Router based routes (`route.ts`), but not ones based on `IncomingMessage` no.

**Pull request merged**